src/main/java/com/epam/ta/reportportal/core/launch/impl/GetLaunchHandler.java

```diff
@@ -102,116 +103,98 @@ public LaunchResource getLaunch(Long launchId, ReportPortalUser.ProjectDetails p
 		 * condition will be added by server(line 66) Due to limitations of the
 		 * com.mongodb.BasicDBObject, user can't add a second 'mode' criteria
 		 */
-		//		validateModeConditions(filter);
-		//			filter = addLaunchCommonCriteria(DEFAULT, filter, projectName);
+		filter = LaunchFilter.of(filter, LaunchModeEnum.DEFAULT);
 		Page<LaunchFull> launches = launchRepository.findByFilter(ProjectFilter.of(filter, projectName), pageable);
+
+		return PagedResourcesAssembler.pageConverter(LaunchConverter.FULL_TO_RESOURCE).apply(launches);
+	}
+
+	public Iterable<LaunchResource> getDebugLaunches(String projectName, ReportPortalUser user, Filter filter, Pageable pageable) {
+
+		filter = LaunchFilter.of(filter, LaunchModeEnum.DEBUG);
+		Page<LaunchFull> launches = launchRepository.findByFilter(ProjectFilter.of(filter, projectName), pageable);
+
+		return PagedResourcesAssembler.pageConverter(LaunchConverter.FULL_TO_RESOURCE).apply(launches);
+	}
+
+	public com.epam.ta.reportportal.ws.model.Page<LaunchResource> getLatestLaunches(String projectName, Filter filter, Pageable pageable) {
+
+		filter = LaunchFilter.of(filter, LaunchModeEnum.DEFAULT);
+		Page<LaunchFull> launches = launchRepository.findByFilter(ProjectFilter.of(filter, projectName), pageable);
```

Comment:
we need to implement here a query that gets only latest launches. it means that launches are grouped by name and in the result only latest (with the highest launch number) launches appear